kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/jdbc/JDBCMetadataStore.scala

```diff
@@ -257,14 +257,15 @@ class JDBCMetadataStore(conf: KyuubiConf) extends MetadataStore with Logging {
       filter: MetadataFilter,
       from: Int,
       size: Int,
-      desc: Boolean = false): Seq[Metadata] = {
+      desc: Boolean = false,
+      orderByKeyId: Boolean = true): Seq[Metadata] = {
     val queryBuilder = new StringBuilder
     val params = ListBuffer[Any]()
     queryBuilder.append("SELECT ")
     queryBuilder.append(METADATA_COLUMNS)
     queryBuilder.append(s" FROM $METADATA_TABLE")
     queryBuilder.append(s" ${assembleWhereClause(filter, params)}")
-    queryBuilder.append(" ORDER BY key_id ")
+    queryBuilder.append(" ORDER BY ").append(if (orderByKeyId) "key_id " else "create_time ")
```

Comment:
have you tried without `ORDER BY` clause? if the result ordering is stable (for example, ordered by pk), it might be the fastest approach

Reply:
It is not guaranteed.

<img width="921" alt="image" src="https://github.com/user-attachments/assets/e5f09198-c2e2-410a-b17a-a66794c00611">

Reply:
But for small dataset, it looks good. I think we can have a try.
<img width="1166" alt="image" src="https://github.com/user-attachments/assets/ecdd76c6-b171-4b3e-9889-24e64a49e0cf">

Reply:
And the UT can not pass without `order by`.